tensorflow/python/ops/ctc_ops.py

```diff
@@ -328,6 +338,18 @@ def ctc_greedy_decoder(inputs, sequence_length, merge_repeated=True):
         sequence found, the negative of the sum of the greatest logit at each
         timeframe.
   """
+
+  if blank_index is not None:
+
+    if blank_index < 0:
+      blank_index += _get_dim(inputs, 2)
+
+    if blank_index != _get_dim(inputs, 2) - 1:
+      part_before = inputs[:, :, :blank_index]
+      part_after = inputs[:, :, blank_index + 1:]
+      part_blank = inputs[:, :, blank_index:blank_index + 1]
+      inputs = array_ops.concat([part_before, part_after, part_blank], axis=2)
```

Comment:
This is a lot of data copying/manipulation in Python, and could have a significant performance impact for users who pass in this arg. Would it be better to do this in the op itself, or in another way that doesn't have a negative effect on performance?

Reply:
Of course! But I thought the way I implemented it, was the same (if I am right) to the ctc_loss and that wouldn't have enough impact to user performance. Despite that, I want to improve it.

Reply:
To improve this; you'll want to update the CTCGreedyDecoderOp under `core/kernels/ctc_decoder_ops.cc` and `core/ops/ctc_ops.cc`

Reply:
@ebrevdo, thank you very much!